crates/tabby-download/src/lib.rs

```diff
@@ -41,49 +85,62 @@ async fn download_model_impl(
                 "Checksum doesn't match for <{}/{}>, re-downloading...",
                 registry.name, name
             );
-            fs::remove_file(&model_path)?;
+
+            fs::remove_dir_all(registry.get_model_store_dir(name))?;
```

Comment:
this implementation is wrong, we need to delete the entire model dir, not just `.../ggml`.